connector_search_engine/models/se_binding.py

```diff
@@ -82,28 +89,17 @@ def write(self, vals):
         return res
 
     def unlink(self):
-        for record in self:
-            if record.sync_state == "new" or (
-                record.sync_state == "done" and not record.active
-            ):
-                continue
-            if record.active:
-                raise UserError(record._msg_cannot_delete_active())
-            else:
-                raise UserError(record._msg_cannot_delete_not_synchronized())
-        return super(SeBinding, self).unlink()
-
-    def _msg_cannot_delete_active(self):
-        return (
-            _("You cannot delete the binding '%s', unactivate it first.")
-            % self.display_name
-        )
-
-    def _msg_cannot_delete_not_synchronized(self):
-        return (
-            _("You cannot delete the binding '%s', wait until it's synchronized.")
-            % self.display_name
-        )
+        to_delete = self.filtered(lambda rec: rec._is_indexed())
+        obsolete_vals_list = [
+            {
+                "index_id": rec.index_id.id,
+                "res_id": rec.id,
+            }
+            for rec in to_delete
+        ]
+        if obsolete_vals_list:
+            self.env["se.index.obsolete.binding"].sudo().create(obsolete_vals_list)
```

Comment:
```suggestion
            self.env["se.binding.todelete"].sudo().create(obsolete_vals_list)
```
IMO we could avoid having another model for this (eg: use a Serialized field on the se_backend).
OTOH if we delete the index we get the cascade for free... Bah, ok :smile:

Reply:
Also I figured Serialized would probably cause concurrent update issues, that a simple insert wouldn't